services/py-api/src/database/repository/teams_repository.py

```diff
@@ -96,3 +96,15 @@ async def get_verified_registered_teams_count(self) -> int:
         # Ignoring mypy type due to mypy err: 'Returning Any from function declared to return "int"  [no-any-return]'
         # which is not true
         return await self._collection.count_documents({"is_verified": True})  # type: ignore
+    
+    async def fetch_team_by_name(self, team_name: str) -> Result:
+        """Fetches a team by the team_name from the participant"""
+        try:
+            team = await self._collection.find_one({"name": team_name})
+            
+            if team is None:  # If no team is found, return an Err
+                return Err(Exception(f"Team with name '{team_name}' not found"))
+
+            return Ok(team)
+        except Exception as e:
+            return Err(e)
```

Comment:
Use a similar pattern when returning the team and the general exception: 
```Python
async def delete(
        self, obj_id: str, session: Optional[AsyncIOMotorClientSession] = None
    ) -> Result[Team, TeamNotFoundError | Exception]:
        """
        Deletes the team which corresponds to the provided object_id
        """
        try:
            LOG.debug("Deleting team...", team_obj_id=obj_id)
            """
            According to mongodb docs result is of type _DocumentType:
            https://pymongo.readthedocs.io/en/4.8.0/api/pymongo/collection.html#pymongo.collection.Collection.find_one_and_delete
            _id is projected because ObjectID is not serializable.
            We use the Team data class to represent the deleted participant.
            """
            result = await self._collection.find_one_and_delete(filter={"_id": ObjectId(obj_id)}, projection={"_id": 0})
            """
            The result is None when the team with the specified ObjectId is not found
            """
            if result:
                return Ok(Team(id=obj_id, **result))
            return Err(TeamNotFoundError())
        except Exception as e:
            LOG.exception("Team deletion failed due to err {}".format(e))
            return Err(e)
```

Reply:
Fixed in commit a9a6b60a1c533cba63bd89fc5bee4362218c00f7